devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -777,6 +781,16 @@ void HumanStateProvider::run()
                                         pImpl->baseVelocitySolution.getVal(4),
                                         pImpl->baseVelocitySolution.getVal(5)};
     }
+    {
+        iDynTree::KinDynComputations* kindyncomputations = pImpl->kinDynComputations.get();
+        pImpl->solution.CoMPosition = {kindyncomputations->getCenterOfMassPosition().getVal(0),
+                                       kindyncomputations->getCenterOfMassPosition().getVal(1),
+                                       kindyncomputations->getCenterOfMassPosition().getVal(2)};
+
+        pImpl->solution.CoMVelocity = {kindyncomputations->getCenterOfMassVelocity().getVal(0),
+                                       kindyncomputations->getCenterOfMassVelocity().getVal(1),
+                                       kindyncomputations->getCenterOfMassVelocity().getVal(2)};
+    }
```

Comment:
If I'm not wrong (maybe @diegoferigo can help), it would be better if all the kindyn calls are moved before the [`std::lock_guard<std::mutex>`](https://github.com/robotology/human-dynamics-estimation/pull/121/files#diff-af5a41cc9a872c1dd7da746d7ad6d0f7R758) block. You can store the output of `kindyncomputations->getCenterOfMassPosition()` and `kindyncomputations->getCenterOfMassVelocity()` into  two buffer variables `CoMPosition ` and `CoMVelocity`. Then the values of the buffer variables is copied into `pImpl->solution.CoMPosition` and `pImpl->solution.CoMVelocity` inside the `std::lock_guard<std::mutex>` block.